include/graphics/texture.h

```diff
@@ -1,16 +1,25 @@
+#pragma once
+
+#include "graphics/primitives.h"
+
 namespace arch {
     
-//typedef unsigned int TextureId;
-using TextureId = unsigned int;
-
+/**
+ * A handle to 2-dimentional 32-bit RGBA image stored in VRAM.
+*/
 class Texture {
 public:
-    Texture(Texture &other) = delete;
-    Texture(Texture &&other) = delete;
-    Texture& operator=(Texture &other) = delete;
-    Texture& operator=(Texture &&other) = delete;
+    Texture() {}
+    Texture(uint32_t width, uint32_t height, const void* data);
```

Comment:
We should avoid using raw pointers to data in our interfaces. Maybe it will be better to pass loader or encapsulated data here?

Reply:
Why though? Texture does not take ownership of data anyway and just copies it to GPU. Passing loader would require making loader interface awkard to use, because then it would have to return void data pointer instead of Texture.

Reply:
I will probably be wrapping the data stored in RAM in next PRs

Reply:
Ok, you're right. That's the best solution.